### PR TITLE
[FEATURE] Create ReloadableExtension interface

### PR DESCRIPTION
**Pull request by nrakhay**
### Description
Extension points (see https://github.com/opensearch-project/opensearch-sdk-java/issues/315) will be (optionally) implemented on Extensions. Eventually these points may need to be communicated to OpenSearch to be registered in the appropriate service or module.

Allowing extensions to inherit from appropriate interfaces depending on their type will match the current Plugin implementation.

### Issues Resolved
Added a new interface that was copied from ReloadablePlugin.java

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/sdk/ReloadableExtension.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sdk;
+
+import org.opensearch.common.settings.Settings;
+
+/**
+ * An extension point for {@link Plugin}s that can be reloaded. There is no
```
Comment (dbwiddis) [inline src/main/java/org/opensearch/sdk/ReloadableExtension.java:15]: Replace "Plugin" with "Extension" in the docs here and elsewhere.

**Review by dbwiddis — COMMENTED** [review]
Looks good other than renaming plugin to extension in the docs.

**Comment by nrakhay** [comment]
Hi!
Could you please check the recent commit?
@dbwiddis @hyandell

**Review by dbwiddis — APPROVED** [review]
Looks great, thanks!    Since this is an abstract method there's nothing we can really test.

**Comment by nrakhay** [comment]
@hyandell @ryanbogan @saratvemulapalli @joshpalis 

Could you please review the latest commit?

**Review by joshpalis — APPROVED** [review]

**Pull request merged**